### ISSUE TITLE
[Play All] should play the full video instead of creating a playlist

### ISSUE THREAD

**Issue by alexrashed** [s0faking/plugin.video.orftvthek#87]
Currently, `base.py#createPlayAllItem` creates a playlist of all items.
https://github.com/s0faking/plugin.video.orftvthek/blob/c23c41a80cec3755fa2c09026cafdfb26bb979af/resources/lib/base.py#L38-L43

This leads to a short interruption (for loading / buffering) when the next item is about to be played.
There also is a full-video url (which would need to be parsed reliably) for each video.

In this issue I wanted to ask if it would be preferred to use a full video-link instead of a playlist to avoid these interruptions. The downside would obviously be that one could not switch between the items of the playlist anymore.

What's your opinion on that? I could probably work on a PR, but I wanted to make sure it's in the interest of the project beforehand.

PS.: Thanks for the awesome plugin! It's a daily driver on my Kodi :)

**Comment by killerbees19** [comment]
If you'd implement a configuration option for it: Why not?

**Comment by s0faking** [comment]
hey @alexrashed,

awesome. i wasn't aware of that streaming option. as @killerbees19 stated i would suggest we add that with a settings option. please feel free to hit me up with a PR.

**Comment by s0faking** [comment]
the gapless video option is now also available for the htmlscraper (in the python3 branch for now). i've changed the logic a bit. the play gapless video listitem is always visible if available. if the setting for the "play all" item is set, the chapter play listitem is shown additionaly

**Comment by killerbees19** [comment]
btw: I've not tested this yet, but I've implemented the gapless feature at a non-public (PHP) script too. And while testing the changes I've found a show without a gapless video file in the JSON…

`Matinee am Sonntag` (03.11.2019) has no progressive gapless video files. Maybe there are other episodes too, I don't know.

```
  ["is_gapless_austria"]=>
  bool(true)
  ["is_gapless"]=>
  bool(false)
  ["sources"]=>
  array(5) {
    ["hls"]=>
    array(0) {
    }
    ["hds"]=>
    array(0) {
    }
    ["smooth"]=>
    array(0) {
    }
    ["dash"]=>
    array(0) {
    }
    ["progressive_download"]=>
    array(0) {
    }
  }
```

I would recommend to check `is_gapless` before attempting to add the (non-existing) gapless video. If you're already doing something similar in the plugin, sorry for the noise! :-)
